### ISSUE TITLE
voltage sensors no work fine

### ISSUE THREAD

**Issue by pipes80-collab** [Fred78290/nct6687d#36]
this si my problem with this module:
dram voltage not reading fine
nct6687-isa-0a20
Adapter: ISA adapter
+12V:           12.19 V  (min = +12.19 V, max = +12.22 V)
+5V:             5.03 V  (min =  +5.03 V, max =  +5.04 V)
+3.3V:           3.34 V  (min =  +3.34 V, max =  +3.34 V)
CPU Soc:         1.11 V  (min =  +1.11 V, max =  +1.11 V)
CPU Vcore:     674.00 mV (min =  +0.67 V, max =  +0.68 V)
CPU 1P8:         0.00 V  (min =  +0.00 V, max =  +0.00 V)
CPU VDDP:        0.00 V  (min =  +0.00 V, max =  +0.00 V)
DRAM:            2.24 V  (min =  +2.24 V, max =  +2.24 V)
Chipset:       994.00 mV (min =  +0.99 V, max =  +0.99 V)
CPU Fan:          0 RPM  (min =    0 RPM, max =    0 RPM)
Pump Fan:      4411 RPM  (min = 4411 RPM, max = 4687 RPM)
System Fan #1:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #2:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #3:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #4:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #5:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #6:    0 RPM  (min =    0 RPM, max =    0 RPM)
CPU:            +27.0°C  (low  = +27.0°C, high = +28.0°C)
System:         +25.0°C  (low  = +25.0°C, high = +26.0°C)
VRM MOS:        +30.0°C  (low  = +30.0°C, high = +30.0°C)
PCH:            +21.0°C  (low  = +21.0°C, high = +21.0°C)
CPU Socket:     +25.0°C  (low  = +25.0°C, high = +25.0°C)
PCIe x1:        +26.0°C  (low  = +26.0°C, high = +26.0°C)
M2_1:           +26.0°C  (low  = +26.0°C, high = +26.0°C)

**Comment by Fred78290** [comment]
> this si my problem with this module:
> 
> dram voltage not reading fine
> 
> nct6687-isa-0a20
> 
> Adapter: ISA adapter
> 
> +12V:           12.19 V  (min = +12.19 V, max = +12.22 V)
> 
> +5V:             5.03 V  (min =  +5.03 V, max =  +5.04 V)
> 
> +3.3V:           3.34 V  (min =  +3.34 V, max =  +3.34 V)
> 
> CPU Soc:         1.11 V  (min =  +1.11 V, max =  +1.11 V)
> 
> CPU Vcore:     674.00 mV (min =  +0.67 V, max =  +0.68 V)
> 
> CPU 1P8:         0.00 V  (min =  +0.00 V, max =  +0.00 V)
> 
> CPU VDDP:        0.00 V  (min =  +0.00 V, max =  +0.00 V)
> 
> DRAM:            2.24 V  (min =  +2.24 V, max =  +2.24 V)
> 
> Chipset:       994.00 mV (min =  +0.99 V, max =  +0.99 V)
> 
> CPU Fan:          0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> Pump Fan:      4411 RPM  (min = 4411 RPM, max = 4687 RPM)
> 
> System Fan #1:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> System Fan #2:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> System Fan #3:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> System Fan #4:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> System Fan #5:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> System Fan #6:    0 RPM  (min =    0 RPM, max =    0 RPM)
> 
> CPU:            +27.0°C  (low  = +27.0°C, high = +28.0°C)
> 
> System:         +25.0°C  (low  = +25.0°C, high = +26.0°C)
> 
> VRM MOS:        +30.0°C  (low  = +30.0°C, high = +30.0°C)
> 
> PCH:            +21.0°C  (low  = +21.0°C, high = +21.0°C)
> 
> CPU Socket:     +25.0°C  (low  = +25.0°C, high = +25.0°C)
> 
> PCIe x1:        +26.0°C  (low  = +26.0°C, high = +26.0°C)
> 
> M2_1:           +26.0°C  (low  = +26.0°C, high = +26.0°C)

Hello what is your mobo?

**Comment by pipes80-collab** [comment]
is a MSI z490 unify

**Comment by pipes80-collab** [comment]
up

**Comment by pipes80-collab** [comment]
now voltage cpu soc is not precise

**Comment by Fred78290** [comment]
Did you blacklisted nct6683 module ?

**Comment by pipes80-collab** [comment]
No, it's needed?

**Comment by pipes80-collab** [comment]
> Did you blacklisted nct6683 module ?

it's not loaded when give lsmod

**Comment by Fred78290** [comment]
Yes because the nct6683 is concurrent of nct6687 and probably the nct6683 is installed inplace of mine.
Values coming from nct6683 are lazy.

**Comment by pipes80-collab** [comment]
I don't know, your module is installed, and nct6683 is not loaded in modules kernel

**Comment by Fred78290** [comment]
As is, the module sample values in the nct6687 register every 100ms nothing else.
So i don't know where come the erroneous values.

Did you fill something in /etc/sensors3.conf or /etc/sensors.conf or checkit ?

**Comment by pipes80-collab** [comment]
> Così com'è, i valori di campionamento del modulo nel registro nct6687 ogni 100 ms nient'altro. Quindi non so da dove vengono i valori errati.
> 
> Hai compilato qualcosa in /etc/sensors3.conf o /etc/sensors.conf o checkit ?

no, i have not compiled nothing on sensor3.conf, i have check now and it's not nct6687d present

**Comment by pipes80-collab** [comment]
yes i have same situation, i attached screenshot
![Uploading Schermata del 2022-07-15 00-26-51.png…]()

**Comment by Fred78290** [comment]
Hello, Have you the ability to run W10 and install Hardware monitor using LibreHardwareMonitor and check the concerned voltage.

Regards

**Comment by opal06** [comment]
> Hello, Have you the ability to run W10 and install Hardware monitor using LibreHardwareMonitor and check the concerned voltage.
> 
> Regards

Yes, that should be possible, but it may take me a few days as I'm quite busy at the moment. But hopefully I will be able to get back to you soon with that.

**Comment by Fred78290** [comment]
Thank you

**Comment by pipes80-collab** [comment]
this is screenshot

![Immagine 2022-07-15 193526](https://user-images.githubusercontent.com/56699284/179298618-a320f4e5-9bc1-4b1a-bc81-5d04c29063fa.png)

**Comment by Fred78290** [comment]
Well I have also no linux machine 10850K with an MSI B460M Bazooka having NCT6687D and under linux I have the error.

I can invest now to find the problem

**Comment by opal06** [comment]
So, I got around to run win10 as well and here are some screenshots:
![Load](https://user-images.githubusercontent.com/41026414/179354221-dec2ad17-c01b-49d5-bc8e-f235226e2f26.PNG)

This is LibreHardwareInfo under load. VCore value is different from nct6687 driver, but still wrong, which can be seen when compared to hwinfo:

![hwinfo load](https://user-images.githubusercontent.com/41026414/179354267-debe4d26-cf61-4dad-a21d-c15352060316.PNG)

This matches what's set in the BIOS, so I think hwinfo actually reads correctly.

I hope this can help you. If you need something else, feel free to ask.

**Comment by pipes80-collab** [comment]
Today i'll try in load

**Comment by pipes80-collab** [comment]
i have make a chinebench r23 test for read voltage.
screen librehardwaremonitor attached
![Immagine 2022-07-16 192943](https://user-images.githubusercontent.com/56699284/179369469-54a5ae0c-806d-48c8-9f39-f52d7a3dbefe.png)

**Comment by Fred78290** [comment]
Hello thank you for your effort, I find finally the problem. In fact between AMD motherboard and Intel motherboard the sensor are not connected with the same register.

So the default register declaration is for AMD mobo.

To allow custom voltage configuration I have created a new branch **manual-voltage** and update the README to explain how to.
So before merge the pull request, could you test it ?

Thank you.

**Comment by opal06** [comment]
I will try to get around to it this evening or tomorrow. Thank you so much for your efforts!

**Comment by opal06** [comment]
So, I have built the new branch and installed it. It works fine, now the key is getting the profile correct. I have made a first attempt for the MSI z490i Unify, which I will attach here. I needed to do some computation, so it might be missreporting.

@pipes80-collab If you could test it, that would be great. There is obviously no guaranty that MSI wired the same sensors up between the z490 unify and the z490i, but it's close enough that it is worth a shot. It would be especially great if you could go and set a few manual, fixed vcore voltages in BIOS to see if my calculations are correct across a few voltages. Everything up to 1.4V should be safe, especially for short tests.

Note: As Github limits the upload to certain file types, I added .txt as extension, which should be removed before use.
[Z490i-unify.conf.txt](https://github.com/Fred78290/nct6687d/files/9128482/Z490i-unify.conf.txt)
.

**Comment by Fred78290** [comment]
@opal06 
Try my sensors conf in sensors.d, I think that it work correctly for you mobo.

**Comment by opal06** [comment]
> @opal06 Try my sensors conf in sensors.d, I think that it work correctly for you mobo.

No, unfortunately that does not seem to work properly:

nct6687-isa-0a20
Adapter: ISA adapter
+12V:          145.73 V  (min = +145.73 V, max = +146.30 V)
+5V:            24.90 V  (min = +24.85 V, max = +25.00 V)
VCore:           3.38 V  (min =  +3.37 V, max =  +3.38 V)
DIMM:            1.34 V  (min =  +1.34 V, max =  +1.35 V)
CPU I/O:         0.00 V  (min =  +0.00 V, max =  +0.00 V)
CPU SA:          0.00 V  (min =  +0.00 V, max =  +0.00 V)
+3.3V:           1.25 V  (min =  +1.24 V, max =  +1.25 V)
CPU Fan:        769 RPM  (min =  715 RPM, max =  769 RPM)
Pump Fan:      1996 RPM  (min = 1990 RPM, max = 2000 RPM)
System Fan #1:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #2:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #3:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #4:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #5:    0 RPM  (min =    0 RPM, max =    0 RPM)
System Fan #6:    0 RPM  (min =    0 RPM, max =    0 RPM)
CPU:            +58.0°C  (low  = +40.0°C, high = +65.0°C)
System:         +54.0°C  (low  = +54.0°C, high = +54.0°C)
VRM MOS:        +45.0°C  (low  = +44.0°C, high = +45.0°C)
PCH:            +62.0°C  (low  = +61.0°C, high = +62.0°C)
CPU Socket:     +39.0°C  (low  = +38.0°C, high = +39.0°C)

We can see that both +12V and +5V are too high, which is fixed by removing the compute for them. Same goes for Vcore, which is way too high, while DRAM is correct.
So maybe MSI wired the chip differently from z490 to b460... This might be due to different CRM components. Unfortunately, I cannot find the VRM specs for the b460m Bazooka. The z490i Unify uses 90A smart power stages from Renesas.

**Comment by Fred78290** [comment]
`
+12V: 145.73 V (min = +145.73 V, max = +146.30 V)
+5V: 24.90 V (min = +24.85 V, max = +25.00 V)
 `
Input register look like good just remove the compute multiplier x12 x5 but it means that multipliers applied twice (another sensors conf ? )

` VCore: 3.38 V (min = +3.37 V, max = +3.38 V)`

This input look like +3.3V
` +3.3V: 1.25 V (min = +1.24 V, max = +1.25 V)`

Look like vcore.

**Comment by pipes80-collab** [comment]
> > @opal06 Try my sensors conf in sensors.d, I think that it work correctly for you mobo.
> 
> No, unfortunately that does not seem to work properly:
> 
> nct6687-isa-0a20
> Adapter: ISA adapter
> +12V:          145.73 V  (min = +145.73 V, max = +146.30 V)
> +5V:            24.90 V  (min = +24.85 V, max = +25.00 V)
> VCore:           3.38 V  (min =  +3.37 V, max =  +3.38 V)
> DIMM:            1.34 V  (min =  +1.34 V, max =  +1.35 V)
> CPU I/O:         0.00 V  (min =  +0.00 V, max =  +0.00 V)
> CPU SA:          0.00 V  (min =  +0.00 V, max =  +0.00 V)
> +3.3V:           1.25 V  (min =  +1.24 V, max =  +1.25 V)
> CPU Fan:        769 RPM  (min =  715 RPM, max =  769 RPM)
> Pump Fan:      1996 RPM  (min = 1990 RPM, max = 2000 RPM)
> System Fan #1:    0 RPM  (min =    0 RPM, max =    0 RPM)
> System Fan #2:    0 RPM  (min =    0 RPM, max =    0 RPM)
> System Fan #3:    0 RPM  (min =    0 RPM, max =    0 RPM)
> System Fan #4:    0 RPM  (min =    0 RPM, max =    0 RPM)
> System Fan #5:    0 RPM  (min =    0 RPM, max =    0 RPM)
> System Fan #6:    0 RPM  (min =    0 RPM, max =    0 RPM)
> CPU:            +58.0°C  (low  = +40.0°C, high = +65.0°C)
> System:         +54.0°C  (low  = +54.0°C, high = +54.0°C)
> VRM MOS:        +45.0°C  (low  = +44.0°C, high = +45.0°C)
> PCH:            +62.0°C  (low  = +61.0°C, high = +62.0°C)
> CPU Socket:     +39.0°C  (low  = +38.0°C, high = +39.0°C)
> 
> We can see that both +12V and +5V are too high, which is fixed by removing the compute for them. Same goes for Vcore, which is way too high, while DRAM is correct.
> So maybe MSI wired the chip differently from z490 to b460... This might be due to different CRM components. Unfortunately, I cannot find the VRM specs for the b460m Bazooka. The z490i Unify uses 90A smart power stages from Renesas.

I'll give mu result tonight, After your configuration

**Comment by pipes80-collab** [comment]
I just tested the sensors under load, a comparison with cpu-x, I attach image
![Uploading Schermata del 2022-07-18 21-46-09.png…]()

**Comment by Fred78290** [comment]
> a perfect ignorant question, could there be some error in the numerical arrangement of the sensors? comparing 3 data
> 
> ![Schermata del 2022-07-18 22-56-58](https://user-images.githubusercontent.com/56699284/179616211-f202c6ee-74e4-46d9-81d2-33262153253e.png)

I don't know because the display order is depending of application
